src/run.cpp

```diff
@@ -0,0 +1,192 @@
+#include <stdio.h>
```

Comment:
I might name this something other than run.cpp since it contains much of our core logic and are not just runner scripts

Reply:
Since corels is taken and the main() function is in main.cc, how about "algo.cpp"? I named it run.cpp because it houses the three run functions, run_corels_begin, run_corels_loop, run_corels_end.

Reply:
Ok, we can keep it as is for now